ipaserver/install/upgradeinstance.py

```diff
@@ -237,14 +237,18 @@ def __disable_listeners(self):
 
     def __disable_schema_compat(self):
         ldif_outfile = "%s.modified.out" % self.filename
+
+        with open(self.filename, "r") as in_file:
+            parser = GetEntryFromLDIF(in_file, entries_dn=[COMPAT_DN])
+            parser.parse()
+
+        try:
+            compat_entry = parser.get_results()[COMPAT_DN]
+        except KeyError:
+            return
+
         with open(ldif_outfile, "w") as out_file:
             with open(self.filename, "r") as in_file:
-                parser = GetEntryFromLDIF(in_file, entries_dn=[COMPAT_DN])
-                parser.parse()
-                try:
-                    compat_entry = parser.get_results()[COMPAT_DN]
-                except KeyError:
-                    return
                 parser = installutils.ModifyLDIF(in_file, out_file)
                 if not compat_entry.get('nsslapd-pluginEnabled'):
```

Comment:
You can probably move this check before open(ldif_outfile,...). This way, if the input file does not contain the entry nssldap-pluginEnabled, the outfile doesn't get created and we exit quickly.

Reply:
I will try!

Reply:
Thanks @flo-renaud, 
ipa-server-upgrade is working for me with this change.